api/pkg/template/execute.go

```diff
@@ -187,3 +192,81 @@ func (e *Engine) recordDependency(dependency string) {
 		}
 	}
 }
+
+// Registry template functions
+
+// hasLocalRegistry returns true when local registry deployment exists and is ready
+func (e *Engine) hasLocalRegistry() bool {
+	if e.kubeClient == nil || e.ctx == nil {
+		return false
+	}
+
+	deploy := &appsv1.Deployment{}
+	err := e.kubeClient.Get(e.ctx, client.ObjectKey{
+		Namespace: constants.RegistryNamespace,
+		Name:      "registry",
+	}, deploy)
+
+	return err == nil
+}
+
+// localRegistryHost returns the registry service ClusterIP with port when available
+func (e *Engine) localRegistryHost() string {
+	if !e.hasLocalRegistry() || e.kubeClient == nil || e.ctx == nil {
+		return ""
+	}
+
+	svc := &corev1.Service{}
+	err := e.kubeClient.Get(e.ctx, client.ObjectKey{
+		Namespace: constants.RegistryNamespace,
+		Name:      "registry",
+	}, svc)
+
+	if err != nil || svc.Spec.ClusterIP == "" {
+		return ""
+	}
+
+	return fmt.Sprintf("%s:5000", svc.Spec.ClusterIP)
```

Comment:
I think all the registry values are deterministic and you should not need a kube client and only a service cidr. see the [GetRegistryClusterIP](https://github.com/replicatedhq/embedded-cluster/blob/df68a5584cd6fea844e4e3eacbc5080ff79f6d45/pkg/addons/registry/registry.go#L59) function